include/RAJA/internal/Iterators.hpp

```diff
@@ -105,15 +105,16 @@ class numeric_iterator
   using reference = value_type&;
   using iterator_category = std::random_access_iterator_tag;
 
-  RAJA_HOST_DEVICE constexpr numeric_iterator() : val(0) {}
-  RAJA_HOST_DEVICE constexpr numeric_iterator(const stripped_value_type& rhs)
+  constexpr numeric_iterator() = default;
+  ~numeric_iterator() = default;
+  constexpr numeric_iterator(const numeric_iterator&) = default;
+  constexpr numeric_iterator(numeric_iterator&&) = default;
+  constexpr numeric_iterator& operator=(const numeric_iterator&) = default;
```

Comment:
@trws with these changes, lines 112, 113 generate errors with the Intel 19 compiler (I didn't try other versions): 'invalid type for defaulted assignment operator'.  Code compiles and runs with clang 9 and gcc 8.  Similar for line 281 below, etc.

Also, new warnings: constexpr non-static member function will not be implicitly 'const' in C++14.   The warnings go away with Intel compiler when 'const' is added before '= default', but gnu and clang then give errors about defaulted copy-operators cannot be declared 'const'

Thoughts on what to do here to satisfy all compilers?

Reply:
@ajkunen @willkill07 @mdavis36 if you have ideas about this, please share.

Reply:
i've run into this before.   If you just say `numeric_iterator& operator=(const numeric_iterator&) = default;` i believe that will automatically be constexpr... I think... but one should check the standard, and also check that the compiler agrees

Reply:
the most portable thing to do is:  don't use `default`

Reply:
We can't make them const, but we can parametrize when they are `constexpr` as we have done in other places.  @ajkunen's right that the compiler is supposed to make them `constexpr` automatically if they can be, and explicit `constexpr` is only allowed when it would have been done automatically anyway, so it's basically there to make a compiler complain if it would have made them non-constexpr by mistake.

I can just take `constexpr` out, or make it conditional based on compiler.  This is a case of ICC being buggy, sadly, but clearly we need to work around it one way or another.

Reply:
So removing constexpr on the copy and move assignment satisfies clang, gnu, and intel, and all tests pass. Will test cuda and hip next....

If default doesn't give portability, what's the freakin' point?!? Ay carumba!

Reply:
@trws I'm running some tests now. I will push some changes shortly. Hopefully, this will get the CI builds to pass.

Reply:
Arrrghhh! Spoke too soon. Removing constexpr on the copy and move assignment operators satisfies gnu, clang, nvcc.  Intel generates an internal segmentation violation. 

Now where did I put the contact info for Intel support assistance that it's telling me to report this to....  ❤️

Reply:
> the most portable thing to do is: don't use `default`

Sadly this is true... we may just have to make all these darned things explicit.  It's unfortunate since we only actually need *one* explicit constructor that isn't even from the default set and lose all the defaulted ones to it.  If it wouldn't be an API break I'd even make it aggregate initialized, but it is. 😢